File: app/src/main/java/io/ipoli/android/quest/data/SourceMapping.java
package io.ipoli.android.quest.data;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 5/11/16.
 */
public class SourceMapping {
    private AndroidCalendarMapping androidCalendarMapping;

    private SourceMapping() {
    }

    public static SourceMapping fromGoogleCalendar(long calendarId, long eventId) {
        SourceMapping sourceMapping = new SourceMapping();
        sourceMapping.androidCalendarMapping = new AndroidCalendarMapping(calendarId, eventId);
        return sourceMapping;
    }

    public AndroidCalendarMapping getAndroidCalendarMapping() {
        return androidCalendarMapping;
    }

    public void setAndroidCalendarMapping(AndroidCalendarMapping androidCalendarMapping) {
        this.androidCalendarMapping = androidCalendarMapping;
    }
}

File: app/src/main/java/io/ipoli/android/quest/data/SourceMapping.kt
package io.ipoli.android.quest.data

import io.realm.RealmObject

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 8/19/17.
 */

open class SourceMapping : RealmObject {
    private var androidCalendarMapping: AndroidCalendarMapping? = null

    constructor()

    companion object {

        fun fromGoogleCalendar(calendarId: Long, eventId: Long): SourceMapping {
            val sourceMapping = SourceMapping()
            sourceMapping.androidCalendarMapping = AndroidCalendarMapping(calendarId, eventId)
            return sourceMapping
        }
    }
}

open class AndroidCalendarMapping : RealmObject {
    var calendarId: Long? = null
    var eventId: Long? = null

    constructor()

    constructor(calendarId: Long?, eventId: Long?) {
        this.calendarId = calendarId
        this.eventId = eventId
    }
}